### DISCUSSION TITLE
Drawer closes when Sidebar content gets updated

### DISCUSSION THREAD

**Discussion by MMore** [saadeghi/daisyui#2713] [Q&A]
Hey,
I am using Phoenix Liveview with a drawer. The drawer should show details of specific records. So there is one drawer that I am updating, specifically I am updating the Sidebar Content. After I triggered the update event, I trigger the click event on the checkbox to open the drawer. It seems the update event needs a little longer and the drawer open already. As soon as the update gets applied the drawer closes.

When the content is already up2date (so no DOM change needs to be done), the drawer stays open.

Any good idea to make it work without explicitly waiting for the update to be applied and then opening the drawer?

**Comment by MMore** [comment]
Ok solved it by rendering a `checked` into the checkbox instead of using JS to trigger a click event on the it.

**Reply by MMore** [reply]
Hmm that becomes a problem when the drawer is closed and we try to open it again. That fails because the checkbox is still in checked state.

**Reply by saadeghi** [reply]
Maybe... Instead of triggering `click` the checkbox, you can check or uncheck it using JS?

The drawer simply opens when she checkbox is checked and it will be close when it's unchecked so if you handle the checkbox value correctly, it is expected to work correctly.
